src/main/java/net/minecraftforge/resource/ResourceCacheManager.java

```diff
@@ -69,54 +73,81 @@
         this.pathBuilder = pathBuilder;
     }
 
-    public boolean shouldIndexOffThread()
-    {
-        return !getConfigValue(this.indexOffThreadConfigOption);
+    public ResourceCacheManager(final boolean supportsReloading, final String indexOffThreadConfig, final BiFunction<PackType, String, Path> pathBuilder) {
+        this.supportsReloading = supportsReloading;
+        this.indexOnThreadConfigurationKey = indexOffThreadConfig;
+        this.pathBuilder = pathBuilder;
     }
 
-    public static boolean shouldUseCache()
-    {
-        return getConfigValue(ForgeConfig.COMMON.cachePackAccess);
+    public static boolean shouldUseCache() {
+        return getConfigValue("cachePackAccess", true);
     }
 
-    private static boolean getConfigValue(Supplier<Boolean> configValue)
-    {
-        // Yes we catch the early loading error on purpose.
-        // Both the indexing off-thread and if caching is enabled are configurable, so we need to catch the error.
-        // So when they are loaded early we preserve the default behaviour which is to not cache or index on thread.
-        try
-        {
-            return configValue.get();
-        } catch (IllegalStateException e)
-        {
-            return false;
+    private static boolean getConfigValue(final String configKey, final boolean defaultValue) {
+        final Path configPath = FMLPaths.CONFIGDIR.get().resolve("forge-resource-caching.toml");
+        if (!Files.exists(configPath)) {
+            try {
+                Files.write(configPath, ImmutableList.of(
+                                "# This TOML configuration file controls the resource caching system which is used before the mod loading environment starts.",
+                                "# This file is read by the Forge boot loader, and is not used by the game itself.",
+                                "#",
+                                "# Set this to false to disable the resource cache. This will cause the game to scan the resource packs everytime it needs a list of resources.",
+                                "cacheResources=true",
+                                "",
+                                "# Set this to false to forge the caching of vanilla resources to happen on the main thread.",
+                                "indexVanillaPackCachesOnThread=false",
+                                "",
+                                "# Set this to false to forge the caching of mod resources to happen on the main thread.",
+                                "indexModPackCachesOnThread=false"
+                        ),
+                        StandardOpenOption.CREATE_NEW);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to create the default boot configuration file.", e);
+            }
+        }
+
+        final List<String> lines;
+        try {
+            lines = Files.readAllLines(configPath);
```

Comment:
Yes that is correct 🤔 
It might be worth while to cache this, but I am not 100% on that.

The invocation count is very limited and it is already faster then without caching so not sure.
If I have the time I will look into it, but that would also mean that it would only update after a reboot of the game.

Reply:
I will leave it like this for now.
Invocation count in a typical gameplay loop is minimal (less then 10 if you just start the game and open a world).

This allows hot reloads in case you refresh your resource packs with F3, without requiring additional code to handle it.

If specifically requested I can build it in, but I consider that overkill for the benefit it would get.